### PR TITLE
Fixing string representations in tests

### PR DESCRIPTION
**Pull request by PietropaoloFrisoni**
**Context:** [This PR](https://github.com/PennyLaneAI/pennylane/pull/6542) in core PennyLane slightly improves the string representations for some operators. One example is: `"Adjoint(S(wires=[0]) + T(wires=[0]))"` -> `"Adjoint(S(0) + T(0))"`. Catalyst contains a few tests that check these string representations, so these tests must be updated.

**Description of the Change:** For the above reason, we update string representations for all failing tests.

**Benefits:** Catalyst works again.

**Possible Drawbacks:** None that I can think of.

**Related GitHub Issues:** None.

**Related Shortcut Stories:** [sc-77632]

### CODE REVIEW

File: frontend/test/pytest/test_quantum_control.py [hunk 1]
```diff
@@ -840,16 +840,15 @@ class TestControlledMiscMethods:
 
     def test_repr(self):
         """Test __repr__ method."""
-        assert repr(C_ctrl(qml.S(0), [1])) == "Controlled(S(wires=[0]), control_wires=[1])"
+        assert repr(C_ctrl(qml.S(0), [1])) == "Controlled(S(0), control_wires=[1])"
 
         base = qml.S(0) + qml.T(1)
         op = C_ctrl(base, [2])
-        assert repr(op) == "Controlled(S(wires=[0]) + T(wires=[1]), control_wires=[2])"
+        assert repr(op) == "Controlled(S(0) + T(wires=[1]), control_wires=[2])"
```
Comment (paul0403) [inline frontend/test/pytest/test_quantum_control.py:847]:
```suggestion
        assert repr(op) == "Controlled(S(0) + T(1), control_wires=[2])"
```

**Review by lillian542 — APPROVED** [review]
LGTM once the T gate that slipped through is updated :)

**Comment by paul0403** [comment]
@joeycarter Is this the "changed-before-release-then-changed-back" string test you mentioned?

**Comment by joeycarter** [comment]
> @joeycarter Is this the "changed-before-release-then-changed-back" string test you mentioned?

Yes exactly, this PR replaces my old one here: #1263

**Review by joeycarter — APPROVED** [review]

**Pull request merged**